router/core/operation_parser.go

```diff
@@ -0,0 +1,159 @@
+package core
+
+import (
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/buger/jsonparser"
+	"github.com/cespare/xxhash/v2"
+	"github.com/wundergraph/cosmo/router/internal/pool"
+	"github.com/wundergraph/cosmo/router/internal/unsafebytes"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/ast"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/astnormalization"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/astparser"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/astprinter"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/operationreport"
+)
+
+type ParsedOperation struct {
+	// ID represents a unique-ish ID for the operation calculated by hashing
+	// its normalized representation and its variables
+	ID uint64
+	// Name is the operation name, if any
+	Name string
+	// Type is a string representing the operation type. One of
+	// "query", "mutation", "subscription"
+	Type string
+
+	// Query is the "query" field value in the JSON payload
+	Query string
+	// Variables in the "variables" field value in the JSON payload
+	Variables []byte
+	// NormalizedRepresentation is the normalized representation of the operation
+	// as a string. This is provided for modules to be able to access the
+	// operation.
+	NormalizedRepresentation string
+}
+
+type OperationParser struct {
+	executor     *Executor
+	documentPool *sync.Pool
+}
+
+func NewOperationParser(executor *Executor) *OperationParser {
+	return &OperationParser{
+		executor: executor,
+		documentPool: &sync.Pool{
+			New: func() interface{} {
+				return ast.NewSmallDocument()
+			},
+		},
+	}
+}
+
+func (p *OperationParser) ParseReader(r io.Reader) (*ParsedOperation, error) {
```

Comment:
It's not used anywhere

Reply:
It stopped being used after merging the limited reader PR, I've removed this function